src/native/java_class_ids.h

```diff
@@ -896,6 +896,6 @@ struct java_boxed_array_list_properties {
 };
 extern struct java_boxed_array_list_properties boxed_array_list_properties;
 
-void cache_java_class_ids(JNIEnv *env);
+void aws_cache_jni_ids(JNIEnv *env);
```

Comment:
```suggestion
/**
 * All functions bound to JNI MUST call this before doing anything else.
 * This caches all java class IDs the first time it is called, any further calls are no-op, it is thread-safe.
 * The reason we do this lazily, rather than simply calling it once from awsCrtInit(),
 * is to avoid deadlock when multiple threads init the CRT simultaneously.
 * See: https://github.com/awslabs/aws-crt-java/pull/670
 */
void aws_cache_jni_ids(JNIEnv *env);
```

Reply:
Perfect; thorough and conscientious.

Reply:
It never occurred to me before today to have code link its own PR 🔃🤪🔃

Reply:
Probably a good new best practice.